theseus/core/cost_function.py

```diff
@@ -133,11 +138,21 @@ def __init__(
         # The following are auxiliary Variable objects to hold tensor data
         # during jacobian computation without modifying the original Variable objects
         self._tmp_optim_vars = tuple(v.copy() for v in optim_vars)
-
+        self._tmp_aux_vars = None
         self._tmp_optim_vars_for_loop = None
         self._tmp_aux_vars_for_loop = None
 
-        if autograd_loop_over_batch:
+        self._autograd_loop_over_batch = autograd_loop_over_batch
+        self._autograd_functorch = autograd_functorch
+
+        if autograd_functorch and self._autograd_loop_over_batch:
+            self._autograd_loop_over_batch = False
+            warnings.warn(
+                "autograd_loop_over_batch has been reset to be False due to "
```

Comment:
I don't love this warning message, but I'm struggling to find a good alternative. Any suggestions from @mhmukadam?

I am thinking something like "autograd_use_functorch=True overrides given autograd_loop_over_batch=True, so the latter will be set to False".

Reply:
Yeah what you suggest sounds clearer.